cmd/web.go

```diff
@@ -91,6 +111,9 @@ func runWeb(ctx *cli.Context) error {
 	case setting.HTTP:
 		err = runHTTP(listenAddr, context2.ClearHandler(m))
 	case setting.HTTPS:
+		if setting.RedirectPort80 {
+			go runHTTPRedirector()
```

Comment:
Only if `runHTTP` returns an error. My concern is that this goroutine will stop and then port 80 goes offline. Something needs to keep track of it 🙂